packages/sidecar/src/messageBuffer.spec.ts

```diff
@@ -1,24 +1,371 @@
-import { describe, expect, it } from "vitest";
+import { afterEach, beforeEach, describe, expect, it, vi } from "vitest";
 import { MessageBuffer } from "./messageBuffer.js";
 
 describe("messageBuffer", () => {
-  it("basic usage", async () => {
-    const messageBuffer = new MessageBuffer<number>(5);
+  describe("ring buffer behavior", () => {
+    it("should wrap around when buffer size is exceeded", () => {
+      const messageBuffer = new MessageBuffer<number>(3);
 
-    for (let i = 0; i < 3; i++) {
-      messageBuffer.put(i);
-    }
+      messageBuffer.put(0);
+      messageBuffer.put(1);
+      messageBuffer.put(2);
+      messageBuffer.put(3);
+      messageBuffer.put(4);
 
-    expect(messageBuffer.read()).toEqual([2, 1, 0]);
+      expect(messageBuffer.read()).toEqual([4, 3, 2]);
+    });
+
+    it("should handle exact buffer size fill", () => {
+      const messageBuffer = new MessageBuffer<number>(5);
+
+      for (let i = 0; i < 5; i++) {
+        messageBuffer.put(i);
+      }
+
+      expect(messageBuffer.read()).toEqual([4, 3, 2, 1, 0]);
```

Comment:
what's really he point of this one?

Reply:
Edge case where your incoming message count exactly equals to the buffer size. Ensures you don't have any off-by-one errors anywhere and data is intact even at boundary conditions.